filepicker/src/main/java/dev/arkbuilders/components/filepicker/ArkFilePickerViewModel.kt

```diff
@@ -191,9 +192,8 @@ internal class ArkFilePickerViewModel(
         val rootsWithFavorites = container.stateFlow.value.rootsWithFavs
         val roots = rootsWithFavorites.keys
         val root = roots.find { root -> file.startsWith(root) }
-        val favorites = rootsWithFavorites[root]?.flatten()
 
-        val hasNestedRoot = file.hasNestedOrParentalRoot(roots)
+        val hasNestedRoot = file.hasNestedRoot(roots)
```

Comment:
@mdrlzy By checking if a folder `hasNestedOrParentalRoot`, we are prohibiting pinning current folder if 
- it contains root(s)
- it is being contained by root(s)
By changing from `hasNestedOrParentalRoot` to `hasNestedRoot`, we're ignoring every root folder that might `contain` currently folder, and still allow pining the current folder.

Is that our expectation?

Reply:
If folder has parental root, we should still be able to pin it as favorite. Now this is not possible, check the main branch yourself.
Sorry, my bad, I forgot about it when I was reviewing your PR